sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/Exchange.scala

```diff
@@ -77,6 +77,15 @@ case class ReusedExchangeExec(override val output: Seq[Attribute], child: Exchan
   override def outputOrdering: Seq[SortOrder] = {
     child.outputOrdering.map(updateAttr(_).asInstanceOf[SortOrder])
   }
+
+  override def verboseStringWithOperatorId(): String = {
+    val cdgen = ExplainUtils.getCodegenId(this)
+    val reuse_op_str = ExplainUtils.getOpId(child)
+    s"""
+       |(${ExplainUtils.getOpId(this)}) $nodeName ${cdgen} [Reuses operator id: $reuse_op_str]
```

Comment:
shall we do the same thing for `ReusedSubqueryExec`?

Reply:
@cloud-fan Yeah.. i will need to study a bit on how to exercise this. Let me push with rest of your comments.. I will look into it.

Reply:
@cloud-fan I think we are okay here. Here is a simple plan involving `ReusedSubqueryExec`
```
explain formatted select (select avg(a) from s1) + (select avg(a) from s1) from s1 limit 1;
```
```
== Physical Plan ==
CollectLimit (3)
+- * Project (2)
   +- Scan hive default.s1 (1)


(1) Scan hive default.s1 
Output: []
     
(2) Project [codegen id : 1]
Output    : [(Subquery scalar-subquery#151 + ReusedSubquery Subquery scalar-subquery#151) AS (scalarsubquery() + scalarsubquery())#168]
Input     : []
     
(3) CollectLimit 
Input: [(scalarsubquery() + scalarsubquery())#168]
     
===== Subqueries =====

Subquery:1 Hosting operator id = 2 Hosting Expression = Subquery scalar-subquery#151
* HashAggregate (7)
+- Exchange (6)
   +- * HashAggregate (5)
      +- Scan hive default.s1 (4)


(4) Scan hive default.s1 
Output: [a#160]
     
(5) HashAggregate [codegen id : 1]
Input: [a#160]
     
(6) Exchange 
Input: [sum#171, count#172L]
     
(7) HashAggregate [codegen id : 2]
Input: [sum#171, count#172L]
     
Subquery:2 Hosting operator id = 2 Hosting Expression = ReusedSubquery Subquery scalar-subquery#151
```
Please let me know what you think.

Reply:
So we don't print the query plan for subquery 2. How is it implemented?

Reply:
@cloud-fan Currently its implemented by the following check in `processPlanSkippingSubqueries`. For a change, i have it commented :-)
```
   // ReusedSubqueryExecs are skipped over
    if (plan.isInstanceOf[BaseSubqueryExec]) {
      return startOperatorID
    }
```